### PR TITLE
Refactor: remove unused variable

### PR DESCRIPTION
**Pull request by taozhu-chicago**
#### Problem
Few local `*cost_without_bpf*` variables don't have actual use, possible leftovers from previous changes. Should clean it up before making changes in that area.

#### Summary of Changes
- remove unused variables.

Fixes #

### CODE REVIEW

**Review by steviez — APPROVED** [review]
LGTM (just looks like you need to run `fmt` for CI)

File: ledger/src/blockstore_processor.rs [hunk 1]
```diff
@@ -327,21 +327,17 @@ fn execute_batches(
...
     // Allowing collect here, since it also computes the minimal tx cost, and aggregate cost.
     // These two values are later used for checking if the tx_costs vector needs to be iterated over.
-    // The collection is a pair of (full cost, cost without estimated-bpf-code-costs).
     #[allow(clippy::needless_collect)]
     let tx_costs = sanitized_txs
         .iter()
         .map(|tx| {
             let tx_cost = CostModel::calculate_cost(tx, &bank.feature_set);
             let cost = tx_cost.sum();
-            let cost_without_bpf = tx_cost.sum_without_bpf();
```
Comment (apfitzge) [inline ledger/src/blockstore_processor.rs:340]: Question for potential follow-up: this was the only external use of `sum_without_bpf()`.

Should we consider making the fn not `pub`, or even move the functionality into `sum()`?

File: ledger/src/blockstore_processor.rs [hunk 2]
```diff
@@ -327,21 +327,17 @@ fn execute_batches(
 
     let mut minimal_tx_cost = u64::MAX;
     let mut total_cost: u64 = 0;
-    let mut total_cost_without_bpf: u64 = 0;
     // Allowing collect here, since it also computes the minimal tx cost, and aggregate cost.
     // These two values are later used for checking if the tx_costs vector needs to be iterated over.
-    // The collection is a pair of (full cost, cost without estimated-bpf-code-costs).
```
Comment (apfitzge) [inline ledger/src/blockstore_processor.rs:333]: Thanks for adjusting the comment!

File: ledger/src/blockstore_processor.rs [hunk 3]
```diff
@@ -351,30 +347,24 @@ fn execute_batches(
...
+            if batch_cost >= target_batch_cost || next_index == sanitized_txs.len() {
+                let tx_batch = rebatch_transactions(
+                    &lock_results,
+                    bank,
+                    &sanitized_txs,
+                    slice_start,
+                    index,
+                    &transaction_indexes,
+                );
+                slice_start = next_index;
```
Comment (apfitzge) [inline ledger/src/blockstore_processor.rs:363]: Another comment for potential clean-up:
`slice_start` is `always index + 1`. Can we just move this into the logic of `rebatch_transactions`?

**Review by apfitzge — APPROVED** [review]
Current change looks good - added some comments about other potential clean up I saw

File: ledger/src/blockstore_processor.rs (hunk 1 above)
Comment (taozhu-chicago) [reply ledger/src/blockstore_processor.rs:340]: Ah yes! Will remove `sum_without_bpf` in a separate PR.

File: ledger/src/blockstore_processor.rs (hunk 3 above)
Comment (taozhu-chicago) [reply ledger/src/blockstore_processor.rs:363]: Mmm, the first call of `rebatch_transactions` has slice_start==0, then it goes next "slice". Think current code is good, no?

Comment (apfitzge) [reply ledger/src/blockstore_processor.rs:363]: Yep, you're right - leave it as is.

**Pull request merged**